SleepEngine/Engine/Jobs/Event.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class Event
+{
+    using LockGuard = std::lock_guard<std::mutex>;
+public:
+    Event() = default;
+    Event(Event&&) = default;
+
+    void waitAndReset()
+    {
+        std::unique_lock<std::mutex> lk(m_mutex);

```

Comment:
In C++17, you can avoid specializing template here.
Just `std::unique_lock lk(m_mutex);`

Reply:
cool